torch/_inductor/ir.py

```diff
@@ -4417,6 +4426,8 @@ def __init__(self, sym, data):
             self.sym = sym.args[0]
             self.is_bool = True
 
+        V.graph.symbol_to_dtype[self.sym] = data.get_dtype()
```

Comment:
If we return another data struct, say a data class with symbol and dtype, instead of symbol here, https://github.com/pytorch/pytorch/blob/e309d6fa1c4c3c7fe3f192e40f4aec4b4609c0f5/torch/_inductor/lowering.py#L2558. We need to take care of all the following phases that may expect a symbol type. I am not sure if that's worth to do just for the purpose of this PR.

Reply:
No, I still think this is very suspicious.

Looking over the modifications you made, the change here is solely so that ShapeAsConstantBuffer can be code generated.  But here, there is a strange divergence between the Python in C++. In Python, the shape is code generated as a plain int; however, in C++, the shape is wrapped into a Tensor. This seems... wrong! Now that we are supporting raw integers flowing through AOTInductor, the shape should be represented as is (as a `int64_t`) when flowing through AOTInductor.